pkg/authentication/scram/scram_enabler.go

```diff
@@ -47,9 +61,34 @@ func enableAgentAuthentication(auth *automationconfig.Auth, agentPassword, agent
 	auth.Users = users
 }
 
-func enableDeploymentMechanisms(auth *automationconfig.Auth) {
-	if contains.String(auth.DeploymentAuthMechanisms, scram256) {
-		return
+func enableDeploymentMechanisms(auth *automationconfig.Auth, opts Options) {
+	for _, authMode := range opts.AutoAuthMechanisms {
+		if !contains.String(auth.DeploymentAuthMechanisms, authMode) {
+			auth.DeploymentAuthMechanisms = append(auth.DeploymentAuthMechanisms, authMode)
+		}
+	}
+}
+
+// validateScramOptions validates that all the required fields have
+// a non empty value.
+func validateScramOptions(opts Options) error {
```

Comment:
before we specified valid options directly, now that they are configurable they need to be validated